### ISSUE TITLE
Just a idea: Easy to use PyPi-GUI for everyone...

### ISSUE THREAD

**Issue by jedie** [pypa/packaging-problems#57]
# The problem:

How to distribute a Python end user application to normal users for all platforms?

IMHO there exist no solution that works with a few man-power.

A project can currently to this:
- Unix: Create packages for all distribution and add them to the official repository.
- Windows: Create a installer, a single setup.exe that installs everything (Python interpreter, needed packages)
- MacOS: Don't know...

IMHO only big projects with many man-power can so this, isn't it?
Small projects can provide a more or less complicated README:
- Easy solution: Just say, install with pip... But this waste to the global installation.
- Say, the user should create a virtualenv and install via pip: That are a few steps to do...

There is my self contained bootstrap solution: https://github.com/jedie/bootstrap_env
But this is IMHO not really end user friendly, e.g: non-dev windows users...
# The Idea:

So what's about a PyPi GUI program?
This program must be easy to install themself. But then, it should be very easy to install End-User-Python-Application via a few clicks.

IMHO it should create a virtualenv, install the app via pip and created .desktop/windows-link files for easy startup...

I looked around and didn't find any PyPi-GUI out there...

Any suggestions?

**Comment by gigante** [comment]
@jedie like this https://github.com/perone/stallion ?

**Comment by jedie** [comment]
Stallion seems to be only a webbased viewer for PyPi. You can't say "Install package XY into a new virtualenv", isn't it?

**Comment by gigante** [comment]
...and outdated packages. But... does not install packages. I saw later. Sorry

**Comment by ncoghlan** [comment]
Cross-linking related CPython issue: http://bugs.python.org/issue27051

**Comment by merwok** [comment]
Not sure how useful «just an idea» is. Ideas are easy 🙂

Nothing prevents anyone from writing a PyPI GUI.
There are tools, like GUI editors, that have them: [Mu](https://codewith.mu/en/tutorials/1.2/pypi) is one example, or PyCharm that was already mentioned

The PyPA, or future packaging-council-to-be-named, can’t just order such a tool to exist, nor create one themselves (being already stretched thin maintaining the whole ecosystem). At best they could bless an existing tool and recomment its use in documentation.

(Reference for the GSoC project to create such a tool: https://github.com/python/cpython/issues/71238 / https://github.com/upendra-k14/pip_gui/)

It seems to me that there isn’t something actionable here.
The ticket can be kept open if deemed valuable (stating an issue that _some_ beginners _may_ expect a graphical tool), or closed.  @woodruffw

**Comment by woodruffw** [comment]
Yeah, agreed -- I think this could be valuable, but the structure of PyPA/packaging governance means that this needs to come from an individual maintainer committed to making this a reality, not an edict!
